NewsScraper/DawnCrawler/DawnSpiders/spiders/dawn.py

```diff
@@ -0,0 +1,34 @@
+from DawnSpiders.items import ArticleItem
+from scrapy.linkextractors import LinkExtractor
+from scrapy.spiders import CrawlSpider, Rule
+
+
+class DawnSpider(CrawlSpider):
+    name = 'dawn'
+    allowed_domains = ['dawn.com']
+    start_urls = [
+        "https://www.dawn.com/"
+
+    ]
+    rules = (
+        Rule(LinkExtractor(deny="^((?!/authors/).)*$")),
+        Rule(LinkExtractor(restrict_css="[class^='nav__item nav__item--']"), follow=True),
+        Rule(LinkExtractor(restrict_css='.story'), callback='parse_items', follow=True)
+    )
+
+    def parse_items(self, response):
+        article = ArticleItem()
+        article['title'] = response.css(".story__link::text").get().replace("\n", " ")
+        article['cover_image_url'] = response.css(".media--uneven img::attr(src)").get()
+        article['image_urls'] = response.css(".media--center ::attr(src)").getall()
+        article['published_time'] = response.xpath("//*[@property='article:published_time']/@content").get()
+        article['updated_time'] = response.css(".timestamp--time::text").getall()
+        article['authors'] = response.xpath("//*[@name='author']/@content").get()
+        article['tweets'] = response.css(".Twitter-tweet a::attr(href)").getall()
+        user = response.css(".comment__author::text").getall()
+        comment = response.css(".comment__body p::text").getall()
+        mapped = zip(user, comment)
```

Comment:
`mapped` is not a good variable name and this variable is not needed. Also we still need to loop over user and comment and create proper structure.